patches/roslyn-analyzers/0001-WIP-for-source-build.patch

```diff
@@ -0,0 +1,1191 @@
+-		{0F68246D-5645-486C-9CED-A8C8A97996DE} = {8E48BD4D-5178-47E1-A850-76B5DA2EB9D0}
+ 		{A94F29CC-56BF-4CF6-93B1-3BB13725B776} = {8D64F6D4-951C-413E-8A53-FB708F9FB5CD}
+ 		{FF1C6979-CD73-46DC-94BF-A6D59A252B66} = {8D64F6D4-951C-413E-8A53-FB708F9FB5CD}
+ 		{FCA9533D-04BB-4654-94A7-B9B2065CE0EF} = {8D64F6D4-951C-413E-8A53-FB708F9FB5CD}
+-		{697FE592-9EB8-493C-87DA-6C218A3B113D} = {8D64F6D4-951C-413E-8A53-FB708F9FB5CD}
+ 		{5AFCA3A1-C780-4E44-BFFA-D9224685FCE1} = {8D64F6D4-951C-413E-8A53-FB708F9FB5CD}
+ 		{50FB229D-38DB-4A34-8588-D33671AB68DE} = {C0B86774-8307-444F-9EE4-98D62C3424F9}
+ 		{CADE5623-CC57-43A2-9D2A-B7AB29225406} = {50FB229D-38DB-4A34-8588-D33671AB68DE}
+@@ -470,12 +378,10 @@ Global
+ 		{1C62A2F6-7A44-4030-9D1A-7E7A966736E5} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+ 		{DC277D4D-9479-4B55-95BB-9FDB8A682DE1} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+ 		{9F4C5E8F-10EE-4193-BB36-A2FDE07B5557} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+-		{F4F111E0-7108-47A7-97A1-A5B10207C0D8} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+ 		{C75555DC-D059-4665-A421-4EF5F30BD9B6} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+ 		{8AC54A99-5655-435B-8417-BF2F995BE66A} = {31F86414-C34B-462A-AFB5-BDA7724904F3}
+ 		{02F88961-A08F-451A-93AB-328792A32EBC} = {C0B86774-8307-444F-9EE4-98D62C3424F9}
+ 		{C3A4699A-4701-4EC9-9EE9-4CAB4C9B1A4E} = {02F88961-A08F-451A-93AB-328792A32EBC}
+-		{813C80EB-D614-4258-BD07-42660BEB738A} = {02F88961-A08F-451A-93AB-328792A32EBC}
+ 		{B969F316-815C-47B9-82B3-67FDD9ED694F} = {1A3624C9-EB8A-4F27-98F8-9FAEAB30604E}
+ 		{3F998217-4174-4AD0-A015-A9BFFADB075F} = {1A3624C9-EB8A-4F27-98F8-9FAEAB30604E}
+ 		{64D70469-18B0-446C-80E9-A7C45A747B3E} = {02F88961-A08F-451A-93AB-328792A32EBC}
+diff --git a/build.sh b/build.sh
+new file mode 100755
+index 000000000..84a7478bf
+--- /dev/null
++++ b/build.sh
+@@ -0,0 +1,16 @@
++#!/usr/bin/env bash
++
++source="${BASH_SOURCE[0]}"
++
++# resolve $SOURCE until the file is no longer a symlink
++while [[ -h $source ]]; do
++  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
++  source="$(readlink "$source")"
++
++  # if $source was a relative symlink, we need to resolve it relative to the path where the
++  # symlink file was located
++  [[ $source != /* ]] && source="$scriptroot/$source"
++done
++
++scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
++"$scriptroot/eng/common/build.sh" --restore --build $@
+diff --git a/eng/GenerateAnalyzerNuspec.targets b/eng/GenerateAnalyzerNuspec.targets
+index bcc2bb261..3368b2cdd 100644
+--- a/eng/GenerateAnalyzerNuspec.targets
++++ b/eng/GenerateAnalyzerNuspec.targets
+@@ -22,6 +22,11 @@
+     <PackageLicenseExpression Condition="'$(PackageLicenseExpression)' == ''">Apache-2.0</PackageLicenseExpression>
+   </PropertyGroup>
+ 
++  <PropertyGroup>
++    <DotNetExecutible Condition="'$(OS)' == 'Windows_NT'">$(DotNetRoot)\dotnet.exe</DotNetExecutible>
++    <DotNetExecutible Condition="'$(DotNetExecutible)' == ''">$(DotNetRoot)/dotnet</DotNetExecutible>
```

Comment:
(Nit) `DotNetExecutible` => `DotNetExecutable`

But come to think of it, this might be provided already by `$(DOTNET_HOST_PATH)`... can confirm in the binlog.

No need to reset CI just for this though.